### PR TITLE
Add documentation for Agent attributes

### PR DESCRIPTION
**Pull request by ocaballeror**
In response to #200

### CODE REVIEW

File: osbrain/agent.py [hunk 1]
```diff
@@ -204,6 +214,19 @@ class Agent():
         When set to `True`, the agent will continue executing the main loop.
     running : bool
         Set to `True` if the agent is running (executing the main loop).
+    _async_req_uuid : dict
+        Stores a unique identifier for every REQ or PULL connection.
+    _async_req_handler : dict
+        Stores the handler for every active REQ or PULL connection.
```
Comment (Peque) [inline osbrain/agent.py]: I think these `_async_req_*` are incorrect, but would need to have a look at the code to know what they are exactly doing. Maybe it is not the best name, as they were probably initially created for ASYNC-REQ channels but are now used for SYNC-PUB channels as well...

File: osbrain/agent.py [hunk 2]
```diff
@@ -204,6 +214,19 @@ class Agent():
...
+    _kill_now : bool
+        Used internally to synchronously shut down the agent.
```
Comment (Peque) [inline osbrain/agent.py]: I'd say "During shutdown, this attribute is set for the agent to be killed.`.

File: osbrain/agent.py [hunk 3]
```diff
@@ -204,6 +214,19 @@ class Agent():
...
+    _DEBUG : bool
+        Whether to print debug level messages.
+    _pending_requests : dict
+        Stores the handler for every REQ message sent while waiting for the
+        reply.
```
Comment (Peque) [inline osbrain/agent.py]: I'd say "Stores pending (without reply) asynchronous requests. The asynchronous request UUID is used as key and the handler is set as value." Note that "asynchronous requests" are not the same as REQ messages. Need to add documentation on channels, which is currently missing... :sweat:

File: osbrain/agent.py [hunk 4]
```diff
@@ -196,6 +202,10 @@ class Agent():
     handler : dict
         A dictionary in which the key is the socket and the values are the
         handlers for each socket.
+    context : zmq.Context
+        Used internally to create ZMQ socket objects.
+    poller : zmq.Poller
+        Used internally to wait for incoming data from sockets.
```
Comment (Peque) [inline osbrain/agent.py]: These will be made private soon, so maybe no need to add the "Used internally". In example:

- ZMQ context to create ZMQ socket objects.
- ZMQ poller to wait for incoming data from sockets.

Comment (ocaballeror) [reply osbrain/agent.py]: Yeah, I planned to change those  descriptions when the attributes are made private.

File: osbrain/agent.py (hunk 1 above)
Comment (ocaballeror) [reply osbrain/agent.py]: This is all I could guess from reading the code. If this is not what they are really used for, I will need a more detailed description from you 😄

File: osbrain/agent.py (hunk 3 above)
Comment (ocaballeror) [reply osbrain/agent.py]: That would definitely be helpful for some of us 👍 

There are still a couple of things that are not fully clear to me.

File: osbrain/agent.py (hunk 1 above)
Comment (Peque) [reply osbrain/agent.py]: Maybe:

- Stores the UUIDs of the asynchronous request sockets (used in communication channels).
- Stores the handler for every asynchronous request sockets (used in communication channels).

File: osbrain/agent.py (hunk 3 above)
Comment (Peque) [reply osbrain/agent.py]: Yeah, will try to do that today/tomorrow.

File: osbrain/agent.py (hunk 1 above)
Comment (ocaballeror) [reply osbrain/agent.py]: Yeah, I'll definitely need to read a bit more on how channels work here.

**Pull request merged**